### PR TITLE
Pimp up Guest Lobby

### PR DESCRIPTION
**Pull request by mrkeksi**
Maybe as a transition a solution, since I don't know how to assign the json language files in PHP.

image

### CODE REVIEW

**Comment by ffdixon** [comment]
I love the title :-).

The current page is English only, but we ultimately figure out how to localize this page as well.

**Comment by antobinary** [comment]
@mrkeksi  Thank you for your contribution! Could you please confirm you have filled out a CLA? https://docs.bigbluebutton.org/support/faq.html#why-do-i-need-to-sign-a-contributor-license-agreement-to-contribute-source-code ?

**Comment by mrkeksi** [comment]
> @mrkeksi Thank you for your contribution! Could you please confirm you have filled out a CLA? https://docs.bigbluebutton.org/support/faq.html#why-do-i-need-to-sign-a-contributor-license-agreement-to-contribute-source-code ?

It's just been sent.

**Comment by antobinary** [comment]
Thank you - we received your CLA!

**Comment by miztaka** [comment]
cool :+1:

**Comment by sommer** [comment]
In principle, I really like the improved look - but I'm worried because, now, the spinner keeps going even if the script has long given up (e.g., displaying "No response from moderator" or erroring out). I believe an even nicer implementation would need active triggering so it *really* only animates as long as the script keeps retrying.

Find a quick mockup here:
<https://gist.github.com/sommer/c785c6d606fe165f947ffa75907e216f>

**Comment by mrkeksi** [comment]
Thanks @sommer nice suggestion.
What is the keepSpinnerGoing function for?
Is it desired that the animation is only played 3 times? 
Will implement this next time with a nice canceled icon

**Comment by sommer** [comment]
In my mockup, I’ve been overly cautious:
- the spinner is only shown once the process of trying to get into the room starts. That is, if the process is not really started (client disabled all JavaScript or refused to execute ours) no spinner is shown
- the spinner must be manually restarted (here: by calling keepSpinnerGoing) every ~5 seconds or it stops. This is done whenever getting into the room is retried, so if (because of unforeseen errors, client side restrictions, or for whatever other reason) the retries stop, the spinner automatically stops as well
- when the process is deliberately stopped, the spinner is removed, so we don’t have to rely on the failsafe working

File: bigbluebutton-client/resources/prod/guest-wait.html [hunk 1]
```diff
@@ -1,10 +1,61 @@
...
-<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
-
+<!DOCTYPE html>
+<html>
 <head>
   <title>Guest Lobby</title>
-  <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1" />
-  <style></style>
+  <meta charset="UTF-8">
+  <style>
```
Comment (Tainan404) [inline bigbluebutton-client/resources/prod/guest-wait.html:6]:
```suggestion
  <style>
  	:root {
      --animation-time: 1;
    }
```

File: bigbluebutton-client/resources/prod/guest-wait.html [hunk 2]
```diff
@@ -1,10 +1,61 @@
...
+  .spinner .bounce2 {
+    animation-delay: -0.16s;
+  }
+  .spinner > div {
+    width: 30px;
+    height: 30px;
+    background-color: rgb(255, 255, 255);
+    display: inline-block;
+    border-radius: 100%;
+    animation: 1.4s ease-in-out 0s infinite normal both running sk-bouncedelay;
```
Comment (Tainan404) [inline bigbluebutton-client/resources/prod/guest-wait.html:37]:
```suggestion
    animation: sk-bouncedelay calc(var(--enableAnimation) * 1.4s) infinite ease-in-out both;
```

File: bigbluebutton-client/resources/prod/guest-wait.html [hunk 3]
```diff
@@ -34,7 +85,7 @@
 
 
         if (attempt >= limit) {
-          updateMessage('TIMEOUT_MESSAGE_HERE');
+          updateMessage('No respons from Moderator');
```
Comment (Tainan404) [inline bigbluebutton-client/resources/prod/guest-wait.html:88]:
```suggestion
		  document.documentElement.style.setProperty('--enableAnimation', 0);
          updateMessage('No respons from Moderator');
```

File: bigbluebutton-client/resources/prod/guest-wait.html [hunk 4]
```diff
@@ -60,22 +111,26 @@
         var sessionToken = findSessionToken();
 
         if(!sessionToken) {
-          updateMessage('NO_SESSION_TOKEN_MESSAGE');
+          updateMessage('No session Token received');
```
Comment (Tainan404) [inline bigbluebutton-client/resources/prod/guest-wait.html:114]:
```suggestion
		 	document.documentElement.style.setProperty('--enableAnimation', 0);
          updateMessage('No session Token received');
```

File: bigbluebutton-client/resources/prod/guest-wait.html [hunk 5]
```diff
@@ -60,22 +111,26 @@
...
           return;
         }
 
         pollGuestStatus(sessionToken, 0, ATTEMPT_LIMIT, ATTEMPT_EVERY_MS);
       } catch (e) {
         console.error(e);
-        updateMessage('GENERIC_ERROR_MESSAGE');
+        updateMessage('Error: more details in the console');
```
Comment (Tainan404) [inline bigbluebutton-client/resources/prod/guest-wait.html:121]:
```suggestion
			document.documentElement.style.setProperty('--enableAnimation', 0);
        updateMessage('Error: more details in the console');
```

**Comment by Tainan404** [comment]
Hi @mrkeksi, I add your commit to my PR (#9979), because we removed the package of `bigbluebutton-client`, I also implemented the tweaks that I commented above.

**Comment by lrit18** [comment]
@Tainan404 how can I implement this great style for guest lobby in on bbb server?

**Comment by mrkeksi** [comment]
@lrit18 guest lobby will come in BBB 2.3
If you want to implement it now, I think a development environment is required:
https://docs.bigbluebutton.org/2.2/dev.html

**Comment by miztaka** [comment]
@lrit18 If you are using greenlight, feel free to use my fork where I ported this feature. 
https://github.com/miztaka/greenlight/tree/feature-guest-wait-2.2

**Pull request closed without merging**